modules/codegen/src/main/scala/com/twilio/guardrail/generators/Http4sHelper.scala

```diff
@@ -43,23 +50,19 @@ object Http4sHelper {
             }
           """
 
-    List[Defn](
-      q"""
-        sealed abstract class ${responseSuperType} {
+    val cls = q"""
+        sealed abstract class ${responseSuperType}[..$extraTypeParams] {
           def fold[A](..${foldParams}): A = ${Term.Match(Term.This(Name("")), foldCases)}
         }
       """
-    ) ++ List[Defn](
-      companion
-    )
+    List[Defn](cls, companion)
   }
 
   def generateDecoder(tpe: Type, consumes: Seq[RouteMeta.ContentType]): Term =
     if (consumes.contains(RouteMeta.ApplicationJson))
```

Comment:
looks like it should have just been `consumes.isEmpty || consumes.contains(ApplicationJson)` here, then we can merge this.

Reply:
Modified as requested.

Reply:
Thank you!